src/shader_uniforms.rs

```diff
@@ -0,0 +1,143 @@
+//! Types and methods for setting shader uniforms
+
+
+// External crates.
+use std::ffi::CString;
+use gl;
+use gl::types::{GLboolean, GLint};
+use std::marker::PhantomData;
+
+// Local crate.
+use back_end::GlGraphics;
+
+/// Describes a shader uniform of a given type.
+#[derive(Clone, Copy)]
+pub struct ShaderUniform<T : ?Sized>{
+    location : GLint,
+    phantom : PhantomData<T>,
+}
+
+/// Shader uniform type
+///
+/// For now a small subset
+pub trait UniformType {}
+
+/// Shader uniform float
+pub trait SUFloat : UniformType {}
```

Comment:
Yes a struct does make more sense, I'm fairly new to Rust so it is nice to have a code review.

This is the way I am using it at the moment, the `SUFloat` type gives a layer of static checking.
```
let uniform_time = ctx.get_uniform::<SUFloat>("time").unwrap();
uniform_time.set(ctx, self.time);
```